src/error.rs

```diff
@@ -39,11 +39,10 @@ pub struct Diagnostic {
 }
 
 #[cfg(feature = "use-compiled-tools")]
-impl std::convert::TryFrom<*mut diagnostics::Diagnostic> for Diagnostic {
-    type Error = shared::SpirvResult;
-
-    #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    fn try_from(diag: *mut diagnostics::Diagnostic) -> Result<Self, Self::Error> {
+impl Diagnostic {
+    pub(crate) fn from_diag(
```

Comment:
This needs to be an `unsafe fn`, if the caller is responsible for guaranteeing validity of the incoming pointer.

Reply:
Good point, fixed in a7f201c